watertap/costing/watertap_costing_package.py

```diff
@@ -10,141 +10,31 @@
-from .units.mixer import cost_mixer
-from .units.osmotically_assisted_reverse_osmosis import (
-    cost_osmotically_assisted_reverse_osmosis,
-)
-from .units.pressure_exchanger import cost_pressure_exchanger
-from .units.pump import cost_pump
-from .units.reverse_osmosis import cost_reverse_osmosis
-from .units.uv_aop import cost_uv_aop
-from .units.evaporator import cost_evaporator
-from .units.compressor import cost_compressor
-from .units.heat_exchanger import cost_heat_exchanger
-from .units.electroNP import cost_electroNP
-
-
-class _DefinedFlowsDict(MutableMapping, dict):
-    # use dict methods
-    __getitem__ = dict.__getitem__
-    __iter__ = dict.__iter__
-    __len__ = dict.__len__
-
-    def _setitem(self, key, value):
-        if key in self and self[key] is not value:
-            raise KeyError(f"{key} has already been defined as a flow")
-        dict.__setitem__(self, key, value)
+from idaes.core.base.costing_base import register_idaes_currency_units
 
-    def __setitem__(self, key, value):
-        raise KeyError(
-            "Please use the `WaterTAPCosting.add_defined_flow` "
-            "method to add defined flows."
-        )
-
-    def __delitem__(self, key):
-        raise KeyError("defined flows cannot be removed")
+from watertap.costing.costing_base import WaterTAPCostingBlockData
 
 
 @declare_process_block_class("WaterTAPCosting")
-class WaterTAPCostingData(FlowsheetCostingBlockData):
-    # Define default mapping of costing methods to unit models
-    unit_mapping = {
-        AD: cost_anaerobic_digestor,
-        Mixer: cost_mixer,
-        Pump: cost_pump,
-        EnergyRecoveryDevice: cost_energy_recovery_device,
-        PressureExchanger: cost_pressure_exchanger,
-        ReverseOsmosis0D: cost_reverse_osmosis,
-        ReverseOsmosis1D: cost_reverse_osmosis,
-        OsmoticallyAssistedReverseOsmosis0D: cost_osmotically_assisted_reverse_osmosis,
-        NanoFiltration0D: cost_nanofiltration,
-        NanofiltrationZO: cost_nanofiltration,
-        NanofiltrationDSPMDE0D: cost_nanofiltration,
-        Crystallization: cost_crystallizer,
-        Ultraviolet0D: cost_uv_aop,
-        Electrodialysis0D: cost_electrodialysis,
-        Electrodialysis1D: cost_electrodialysis,
-        ElectroNPZO: cost_electroNP,
-        Electrolyzer: cost_electrolyzer,
-        IonExchange0D: cost_ion_exchange,
-        GAC: cost_gac,
-        Evaporator: cost_evaporator,
-        Compressor: cost_compressor,
-        HeatExchanger: cost_heat_exchanger,
-    }
-
-    def build(self):
-        super().build()
-        self._registered_LCOWs = {}
-
+class WaterTAPCostingData(WaterTAPCostingBlockData):
     def build_global_params(self):
         # Register currency and conversion rates based on CE Index
         register_idaes_currency_units()
 
+        self._build_common_global_params()
+
         # Set the base year for all costs
         self.base_currency = pyo.units.USD_2018
```

Comment:
While we are making major changes to costing, can we change our default currency to be something more recent? 2021 is available as a unit, but probably not a good representative/default costing year since the pandemic made year/year inflation high, but 2020 or 2022 might be better.

Reply:
Right now nothing in this PR changes the underlying models in any way, it's just a giant reshuffling and consolidation of existing code and functionality.

I would suggest putting off a change like this for when we merge the costing packages.